### PR TITLE
[docs] Note that the S3 plugin uses the OS/container truststore

### PR DESCRIPTION
**Pull request by joostdecock**
We were having problems with the configuration of this plugin, since it kept throwing certificate validations errors.
In our case, we were connecting to an S3-compatible endpoint in our own infrastructure, of which the certificate was provided by our own internal CA.
What was confusing is that our internal CA is trusted by Elasticsearch, it's included in the truststore.

However, since this plugin wraps the `AmazonS3Client`, seems that it's using the default truststore/CA certificates, rather than the one configured in elasticsearch.

That sort of makes sense once you figure it out, but seems like that kind of caveat that should at least be documented. So I'm adding it here.

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)? **I have, this is not my first PR**
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)? **I have, this is not my first PR**
- If submitting code, have you built your formula locally prior to submission with `gradle check`? **Not relevant, docs changes only**
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed. **Not relevant, docs changes only**
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)? **Not relevant, docs changes only**
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that. **Not relevant, docs changes only**

### CODE REVIEW

**Comment by original-brownbear** [comment]
@joostdecock thanks for opening this PR, I think the behavior is well worth documenting for sure. I'm a little confused on how this relates to https://github.com/elastic/elasticsearch/issues/64800 ? Isn't what you're documenting here the 6.x behavior or am I misunderstanding something here?

**Comment by joostdecock** [comment]
I must admit I wasn't aware of #64800 but it seems at the very least closely related (note how they also use the S3 plugin in to reproduce this). But I wouldn't call this a bug per se.

We ran into this issue on a cluster running 7.8.1, so it's not some legacy 6.x behavior. Our situation is as follows:

 - We run elasticsearch inside docker
 - We need the S3 plugin, so we have a dockerfile that adds it, using the upstream (your) docker image as base
 - What we've learned is that we need to add the S3 endpoint's CA to the OS trusted root CAs (inside the container) even though it's already included in the truststore that we have configured in Elasticsearch (and that is made available via a mounted volume)

The only alrternative that I see is that Elasticsearch would somehow copy that certificates in the configured truststore to the operating system's (or container) trusted root CAs. But I can imagine just because you want ES to trust something doesn't mean you want the entire OS to.

So I don't think there's anything wrong here. It's just not obvious to figure out why it doesn't work based on the current documentation, so I wanted to improve that. 

It's also worth noting that the S3 plugin needs to be configured in the *elasticsearch keystore*. Which is (in this context) different from the JVM keystore that holds the cluster's certificates.

So there's just a lot of room for misunderstandings here, and I wanted to try avoid that.

To summarize:

> Isn't what you're documenting here the 6.x behavior or am I misunderstanding something here?

No. This happens in current releases (when running inside Docker at least)

**Comment by DaveCTurner** [comment]
This is a good thing to document but it's taken us too long to get around to reviewing this to be able to iterate on the exact wording (the branch is stale and the OP's fork is gone). I've opened #77676 to continue this, using a `Co-authored-by` tag to properly record the provenance of this change, and am closing this.

**Comment by joostdecock** [comment]
Woops, I guess I had forgotten why I had that fork and KonMary'd it out of existence.

Thanks for getting around to it.
